tests/test_isrFunctions.py

```diff
@@ -0,0 +1,402 @@
+
+    def test_trimToMatchCalibBBox(self):
+        r"""Trim to match raw data.
+
+        Expect bounding boxes to match.
+        """
+        darkExp = isrMock.DarkMock().run()
+        darkMi = darkExp.getMaskedImage()
+
+        nEdge = 2
+        newDark = darkMi[nEdge:-nEdge, nEdge:-nEdge, afwImage.LOCAL]
+        darkMi = newDark
+        newInput = ipIsr.trimToMatchCalibBBox(self.mi, darkMi)
+
+        self.assertEqual(newInput.getImage().getBBox(), newDark.getImage().getBBox())
+
+    def test_darkCorrection(self):
+        r"""Dark correction.
+
+        Expect round-trip application to be equal.
+        Expect RuntimeError if sizes are different.
+        """
+        darkExp = isrMock.DarkMock().run()
+        darkMi = darkExp.getMaskedImage()
+
+        image = self.mi.getImage().getArray()
+        ipIsr.darkCorrection(self.mi, darkMi, 1.0, 1.0, trimToFit=True)
+        ipIsr.darkCorrection(self.mi, darkMi, 1.0, 1.0, trimToFit=True, invert=True)
+
+        diff = image - self.mi.getImage().getArray()
+        assert (diff.sum() == 0)
+
+        newDark = darkMi[1:-1, 1:-1, afwImage.LOCAL]
+        darkMi = newDark
+
+        with self.assertRaises(RuntimeError):
+            ipIsr.darkCorrection(self.mi, darkMi, 1.0, 1.0, trimToFit=False)
+
+    def test_biasCorrection(self):
+        r"""Bias correction.
+
+        Expect smaller median image value after.
+        Expect RuntimeError is sizes are different.
+        """
+        biasExp = isrMock.BiasMock().run()
+        biasMi = biasExp.getMaskedImage()
+
+        ipIsr.biasCorrection(self.mi, biasMi, trimToFit=True)
+
+        assert True
+
+        newBias = biasMi[1:-1, 1:-1, afwImage.LOCAL]
+        biasMi = newBias
+
+        with self.assertRaises(RuntimeError):
+            ipIsr.biasCorrection(self.mi, biasMi, trimToFit=False)
+
+    def test_flatCorrection(self):
+        r"""Flat correction.
+
+        Expect round-trip application to be equal
+        Expect RuntimeError if sizes are different or if scaling type isn't known.
+        """
+        flatExp = isrMock.FlatMock().run()
+        flatMi = flatExp.getMaskedImage()
+
+        image = self.mi.getImage().getArray()
+        for scaling in ('USER', 'MEAN', 'MEDIAN', 'UNKNOWN'):
+            with self.subTest(scaling=scaling):
+                if scaling == 'UNKNOWN':
+                    with self.assertRaises(pexExcept.Exception):
+                        ipIsr.flatCorrection(self.mi, flatMi, scaling, userScale=1.0, trimToFit=True)
+                        ipIsr.flatCorrection(self.mi, flatMi, scaling, userScale=1.0,
+                                             trimToFit=True, invert=True)
+                else:
+                    ipIsr.flatCorrection(self.mi, flatMi, scaling, userScale=1.0, trimToFit=True)
+                    ipIsr.flatCorrection(self.mi, flatMi, scaling, userScale=1.0,
+                                         trimToFit=True, invert=True)
```

Comment:
This lower block doesn't test anything.

This is another loop that would be clearer as separate methods, or at least with `UNKNOWN` as a separate method.